Makefile

```diff
@@ -0,0 +1,46 @@
+# Copyright 2024 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Makefile for building a Cloud Armor Node Controller
+#
+
+#  __     __         _       _     _
+#  \ \   / /_ _ _ __(_) __ _| |__ | | ___ ___
+#   \ \ / / _` | '__| |/ _` | '_ \| |/ _ \ __|
+#    \ V / (_| | |  | | (_| | |_) | |  __\__ \
+#     \_/ \__,_|_|  |_|\__,_|_.__/|_|\___|___/
+#
+
+REPOSITORY ?=
+version := 0.1
+server_tag := $(REPOSITORY)/cloud-armor-node-controller:$(version)
+
+#   _____                    _
+#  |_   _|_ _ _ __ __ _  ___| |_ ___
+#    | |/ _` | '__/ _` |/ _ \ __/ __|
+#    | | (_| | | | (_| |  __/ |_\__ \
+#    |_|\__,_|_|  \__, |\___|\__|___/
+#                 |___/
+
+# Build a docker image for the server, and tag it
+build:
+	docker build --tag=$(server_tag) .
+
+push: build
+	docker push $(server_tag)
+
+# build and push the example custom controller image with specified tag
+cloud-build:
+	gcloud builds submit --config=cloudbuild.yaml
```

Comment:
I would add `--substitutions _REPOSITORY=$(REPOSITORY)`, then you can just do `make REPOSITORY=<blah> cloud-build`.

Reply:
Done